src/usb_stream.cpp

```diff
@@ -349,10 +349,52 @@ bool UsbStream::close_device()
 }
 
 bool UsbStream::read_device(std::vector<uint8_t>& msg, const bool stream_on)
+{
+    // read device and then store number of bytes read
+    return_value_ = UsbStream::read_maybe_device(msg, stream_on);
+    /**
+     * Check the potential error:
+     *
+     * - First we check if the port could be read at all.
+     * - Then we check if the port was read before the timeout
+     * - Then we check the validity of the message FIXME where is validity checking?
```

Comment:
FIXME?

Reply:
Oh, this is because I was reading the comments. The comment describe 3 sections: port reading, port timeout, and message validity. However, the code only has port reading at L363 and port timeout at L374. Within this contiguous code chunk, I don't see anything that checks the validity of the message. The original code is structured like this too (https://github.com/machines-in-motion/real_time_tools/blob/master/src/usb_stream.cpp#L433), so it's not something I introduced when I added the new function. I only added the part that says "FIXME where is validity checking?" Should we get rid of the comment that says "Then we check the validity of the message," then?

Reply:
Nevermind, I will do this in a separate pull request.